admin/client.go

```diff
@@ -218,13 +217,22 @@ func (c *Client) ListChangeSummaries(
 		return nil, err
 	}
 
-	newDoc, err := document.NewInternalDocumentFromSnapshot(key, seq, snapshotMeta.Lamport, snapshotMeta.Snapshot)
+	newDoc, err := document.NewInternalDocumentFromSnapshot(
+		key,
+		snapshotMeta.ServerSeq,
+		snapshotMeta.Lamport,
+		snapshotMeta.Snapshot,
+	)
 
 	if err != nil {
 		return nil, err
 	}
 	var summaries []*types.ChangeSummary
 	for _, c := range changes {
+		if c.ServerSeq() <= newDoc.Checkpoint().ServerSeq {
+			continue
+		}
```

Comment:
May I know what situation this code was added to handle?

Reply:
Not required in normal situations, but required if the '--backend-snapshot-with-purging-changes' flag is used!

In the current implementation, the changes that are deleted are before the lowest serverSeq recorded in the syncedseqs collection.
So it is based on the most lastly updated value in syncedseqs, not the changes just before the snapshot.

In this case, changes that are older than the latest snapshot may remain. I added it to skip remained changes.